token/invocation/proof.go

```diff
@@ -58,18 +58,18 @@ func (t *Token) verifyProofs(delegations []*delegation.Token) error {
 
 	cmd := t.command
 	iss := t.issuer
-	aud := t.audience
-	if !aud.Defined() {
-		aud = t.subject
+	sub := t.subject
+	if t.audience.Defined() {
+		sub = t.subject
```

Comment:
This doesn't do anything.  If it did what I think you intended, it would overwrite the `sub` irretrievably.  The `sub` field should still match all the `delegation` token `sub` fields though and we'll need that to check the chain.